HW_5.py

```diff
@@ -73,6 +73,7 @@ def write_article():
 
 
 if __name__ == '__main__':
+    # Please don't forget to create a pull request before merging to main
     write_article()
```

Comment:
This logic works well, but it's not a pythonic way. For example, your call write_article function 2 times and initialize the another_article variable 2 times.
I'd write something like that:
```
user_input = input('Do you want to publish? Y/N:')
        if user_input == 'Y':
            write_article()
        elif user_input == 'N':
            break
        else:
            print("Incorrect input. Please type Y or N\n")
```

Reply:
Instead of
 ```f"\n------------------------------\n\n"```
You can write:
```f"\n{'-'*23}\n\n"```
23 is number of hyphens

Reply:
you initialize the header variable 2 times. Also, values from input_versions don't use for the header.
```
    input_versions = ('News', 'PrivatAdv', 'Consider')
    header = input('Enter the topic you are interested in:\nNews\nPrivatAdv\nConsider\n:')
    while header not in input_versions:
        print('Please select correct topic:')
        header = input('News\nPrivatAdv\nConsider\n:')

```
I'd change it to:
```

    input_versions = ['News', 'PrivatAdv', 'Consider']
    while True:
        user_input = input(f"Enter the topic you are interested in: {', '.join(input_versions)}\n")
        if user_input not in input_versions:
            print('Please select correct topic')
        else:
            break
```

Reply:
The program is crashed if I use an invalid date here
``` expiration_date = list(datefinder.find_dates(input('Till what date we should keep it?\nPlease follow DD-MM-YYYY date version:')))[0].date()```
For example, the result for`13-13-3000`:
![image](https://user-images.githubusercontent.com/34323554/109114860-e5b20f80-7746-11eb-9d2a-71781ddb1701.png)

Reply:
```text = input('Type your text here\n:')```

The default input doesn't support the multiline text. It seems to me that you fixed that in newer homework, but need to check..

Reply:
You can simplify
```
Article.__init__(self, header=header, text=text)
```
To
```Article.__init__(self, header, text)```

Reply:
I'd define format_text() as abstact method. 

More details: http://pythonicway.com/education/python-oop-themes/33-python-abstract-class